pkg/ucp/backend/controller/resourceproviders/apiversion_put.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourceproviders
+
+import (
+	"context"
+	"fmt"
+
+	ctrl "github.com/radius-project/radius/pkg/armrpc/asyncoperation/controller"
+	"github.com/radius-project/radius/pkg/ucp/datamodel"
+)
+
+var _ ctrl.Controller = (*APIVersionPutController)(nil)
+
+// APIVersionPutController is the async operation controller to perform PUT operations on API versions.
+type APIVersionPutController struct {
+	ctrl.BaseController
+}
+
+// Run implements the controller interface.
+func (c *APIVersionPutController) Run(ctx context.Context, request *ctrl.Request) (ctrl.Result, error) {
+	err := c.ensureSummary(ctx, request)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (c *APIVersionPutController) ensureSummary(ctx context.Context, request *ctrl.Request) error {
+	// Ensure that the resource provider summary contains this resource type + api version.
+	//
+	// We need to use opportunistic concurrency control to avoid conflicting writes.
+	// The properties of the summary are also set by other resources.
+	id, summaryID, err := resourceProviderSummaryIDFromRequest(request)
+	if err != nil {
+		return err
+	}
+
+	// Passing 'summaryNotFoundFail' because we expect the summary to already exist.
+	return updateSummaryWithETag(ctx, c.StorageClient(), summaryID, summaryNotFoundFail, func(summary *datamodel.ResourceProviderSummary) error {
+		if summary.Properties.ResourceTypes == nil {
+			summary.Properties.ResourceTypes = map[string]datamodel.ResourceProviderSummaryPropertiesResourceType{}
+		}
+
+		resourceTypeName := id.Truncate().Name()
+		resourceTypeEntry, ok := summary.Properties.ResourceTypes[id.Truncate().Name()]
+		if !ok {
+			// If we get here, the resource type entry doesn't exist! Something is out of whack.
+			return fmt.Errorf("resource type entry %q not found", resourceTypeName)
+		}
+
+		if resourceTypeEntry.APIVersions == nil {
+			resourceTypeEntry.APIVersions = map[string]datamodel.ResourceProviderSummaryPropertiesAPIVersion{}
+		}
+
+		apiVersionName := id.Name()
```

Comment:
Checking, is this right? Ln 60 above was:
`resourceTypeName := id.Truncate().Name()`

Reply:
Yes, this is intended. Truncate will find the parent's name.